### ISSUE TITLE
Session-related metrics

### ISSUE THREAD

**Issue by Myrtle** [WICG/attribution-reporting-api#79]
Hi guys!
Considering that performance-based ads are here to prove that a campaign has a real impact on the activity of an advertiser, could it be possible to include into this API the measurement of sessions-related metrics?
Knowing that a user who clicked on an ad has spent a lot of time on your website and/or visited a lot of pages is considered as a good proxy to measure the performance of this ad.
It is obviously a weaker signal than a conversion but it is still a good indication for advertisers.
Those metrics are often provided by tools like Google Analytics or directly by ad-tech themselves, using tricks like URL decorations like the ones with the UTM parameters, and third-party scripts embedded on advertisers' pages.

So what about sending a report which will give, at minimal, the time spent on a website? This value doesn't have to be precise. We can imagine having a restricted set of values:
- 0 => the user spent very few seconds on the website and/or visited only one page (what is called a bounce today)
- 30 => the user spent more than 30 seconds
- 60 => the user spent more than 1 minute
- 120 => the spent more than 2 minutes
- 300 => the user spent more than 5 minutes

This reporting would be subject to the same rules as the conversion report (e.g only sent in reporting windows every 2 days or so, potentially affected by some noise?) and should be sent even if no conversion happened during the session itself.

To sum up, something like:
- A user clicks on an ad displayed on `fancy-publisher.com`
- He lands on the homepage of `fancy-advertiser.com`, the browser records the timestamp of the landing
- Approx. 3 minutes later, the user leaves `fancy-advertiser.com` to go visiting `another-publisher.com` 
- The browser detects that the domain of the website has changed, computes the time spent on the freshly left domain, and schedule a report.
- 2 days later a report is sent to the `reportorigin`: `https://ad-tech.com/.well-known/user-session?impression-data=12345678&duration=120`

**Comment by csharrison** [comment]
Thanks Myrtle. I see how this information could be useful. In some ways (with very minor changes) I could see this being possible to implement with the existing API, if the "user abandoned site" was the trigger to log a conversion, and you use  the Beacon API to register your conversion.

The biggest challenge in supporting this natively is configuring this channel of data in the API without revealing extra information that could be used to track the user. Remember, the advertiser site can just as easily record how long their users engage with the site and try to use that to join user identity using the API.

**Comment by Myrtle** [comment]
Thank you for your answer @csharrison . Don't you think having approximate, and possibly noised, values for the session duration will prevent an advertiser to cross-check the values he gathered and his side and the values gathered by the API?

**Comment by csharrison** [comment]
> Don't you think having approximate, and possibly noised, values for the session duration will prevent an advertiser to cross-check the values he gathered on his side with the values gathered by the API?

I think it depends on how approximate and how much noise is added. e.g. this would be a lot easier to justify if it came alongside a reduction in the information revealed in conversion metadata. This would naturally be the case if we implemented this with the Beacon API, so you wouldn't get any "extra" information about the user's behavior on your site, and it would be treated like any other generic information you want to log about a conversion.

**Comment by jbpringuey** [comment]
Would the Beacon API solution still send the report with the &duration=120 parameter ? If that is the case, I think it could work. What would that Beacon API call look like ?

**Comment by csharrison** [comment]
In the case I was referring to you would need to encode the duration in the 3 bits of conversion metadata present in the API right now for event-level reports, and with aggregate buckets for aggregate reports. As cross site data, this value should be coarsened / noised just like any other site data.

**Comment by johnivdel** [comment]
This type of measurement is possible today with header based trigger registration, assuming the trigger data bits on event level reports / or the keys in aggregate reports are being configured to measure this signal directly. Note this does not support measuring both duration and other signals outside of the 3 bits of information in the event-level API.
